### PR TITLE
Adding theming documentation

### PR DESCRIPTION
**Pull request by douglaseggleton**
Adding initial documentation for Theming support for Embed.

A small update to the focus ring needs to be applied to embed, support for `shadows.focusRing`

### CODE REVIEW

File: guides/embed/theming.mdx [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+     * Body - Used for all text
+     */
+    body: "google:Lato, Tahoma, Arial";
+  }
+}
+```
+
+:::info Using Google Fonts
+
+Fonts prefixed with `google:` are downloaded automatically from [Google Fonts](https://fonts.google.com/). This allows you to choose the closest font to your brand. We recommend choosing a font with weights: `400`, `600` and `700`.
```
Comment (cbetta) [inline guides/embed/theming.mdx:107]:
```suggestion
Fonts prefixed with `google:` are downloaded automatically from [Google Fonts](https://fonts.google.com/). This allows you to choose the closest font to your brand. We recommend choosing a font with support for the weights `400`, `600` and `700`.
```

File: guides/embed/theming.mdx [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+
+:::
+
+### Colors
+
+Colors are named semanticly e.g. `primary` or `danger`. This means we can theme new features or UI elements without explicitly asking you to update colors. As an example, we may introdce a 'Remove Card' button at a later stage that would be based of the `danger` set of colors.
```
Comment (cbetta) [inline guides/embed/theming.mdx:113]:
```suggestion
Colors are named semanticly, for example `primary` or `danger`. This allows us to update introduce new features or UI elements without explicitly requiring you to update your theme. As an example, we may introduce a 'Remove Card' button at a later stage that would be based on the `danger` set of colors.
```

File: guides/embed/theming.mdx [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+  }
+}
+```
+
+:::info Reference an existing theme option
+
+Existing theme options can be referenced using CSS variables. This object will be flattened to CSS variables, e.g. `colors.primary` will become `--colors-primary` which
+can be used as follows: `0 0 0 3px (--colors-primary)`.
+
+:::
```
Comment (cbetta) [inline guides/embed/theming.mdx:230]: Do we really want to document this? Does this lock us in to using CSS variables?

File: guides/embed/theming.mdx [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+
+## FAQ
+
+### How do I add custom styles?
```
Comment (cbetta) [inline guides/embed/theming.mdx:234]:
```suggestion
### How do I load my own CSS files?
```

Suggestion as I currently think our theme allows you to set your own style.

File: guides/embed/theming.mdx [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+
+We currently do not support adding your own styles directly via CSS.
+This allows us to roll out updates and ensure compatibility with new features.
```
Comment (cbetta) [inline guides/embed/theming.mdx:237]:
```suggestion
Using a theme allows us to roll out updates and ensure compatibility with new features.
```

File: guides/embed/theming.mdx (hunk 3 above)
Comment (douglaseggleton) [reply guides/embed/theming.mdx:230]: I can remove this - It does lock us to this format, you're right. e.g. I can see later on when we move some styles to the merchant page, we will need to namespace these with `--gr4vy-colors-primary`, so probably best to leave this out for the time being.

File: guides/embed/theming.mdx [hunk 6]
```diff
@@ -0,0 +1,245 @@
...
+
+### What browsers are supported?
+
+Theming uses CSS variables under-the-hood and so for browsers that do not support CSS variables (e.g. IE11) the default theme will be used.
```
Comment (cbetta) [inline guides/embed/theming.mdx:241]:
```suggestion
Theming is currently supported in all modern browsers including Microsoft Edge. Some browsers like Internet Explorer 11 are not supported, and in these the default theme will be used.
```

File: guides/embed/theming.mdx [hunk 7]
```diff
@@ -0,0 +1,245 @@
...
+
+### What values can be set?
+
+Values can be a valid CSS property value. e.g. `colors.primary` could be a hex, rgb or hsl. You can also have multiple shadows for a focus ring, e.g. `0 0 0 2px white, 0 0 0 4px green`
```
Comment (cbetta) [inline guides/embed/theming.mdx:245]: Remind me? Do we do anything to stop a user from loading any `url`s?

**Review by cbetta — APPROVED** [review]
Few small suggestions

File: guides/embed/theming.mdx (hunk 3 above)
Comment (cbetta) [reply guides/embed/theming.mdx:230]: Yeah let's remove.

File: guides/embed/theming.mdx (hunk 7 above)
Comment (douglaseggleton) [reply guides/embed/theming.mdx:245]: Yep - There is a strict Content Security Policy, which should only allow requests to their Gr4vy cluster and google fonts.

```
"default-src 'self' https://*.gr4vy.app; img-src https://*.gr4vy.app data:; style-src 'self' https://fonts.googleapis.com; font-src https://fonts.gstatic.com;"
```

This should be enough to mitigate IMO, but we could do some further sanitzation, or stripping of values.

**Pull request merged**